pkgs/by-name/di/dita-ot/package.nix

```
@@ -0,0 +1,47 @@
+{ stdenv
+, fetchzip
+, bash
+, jdk17
+, nix
+, coreutils
+, lib
+}:
+
+stdenv.mkDerivation rec {
+  name = "dita-ot";
+  version = "4.2.1";
+
+  buildInputs = [
+    bash
+    jdk17
+  ];
+
+  src = fetchzip {
+    url = "https://github.com/dita-ot/dita-ot/releases/download/${version}/dita-ot-${version}.zip";
+    hash = "sha256-PtscKsQwW1uBu0OfVHi7/d9JujGx12nZT+W3uMPQQWc=";
+  };
+
+  installPhase = ''
+    runHook preInstall
+
+    mkdir -p $out/
+    cp -r $src/* $out/
+
+    runHook postInstall
+  '';
+
+  doInstallCheck = true;
+  installCheckPhase = ''
+    runHook preInstallCheck
+    $out/bin/dita --version
+    runHook postInstallCheck
+  '';
+
+  meta = {
+    homepage = "https://dita-ot.org";
```

You could also add a `changelog` entry, since there seems to be a release notes page:

```suggestion
    homepage = "https://dita-ot.org";
    changelog = "https://www.dita-ot.org/dev/release-notes/#v${version}"
```